kernel/src/arch/riscv/cpu.rs

```diff
@@ -121,14 +84,67 @@ macro_rules! copy_gp_regs {
     };
 }
 
-impl GpRegs {
-    pub fn copy_to_raw(&self, dst: &mut GeneralRegs) {
-        copy_gp_regs!(self, dst);
+/// Represents the context of a signal handler.
+///
+/// This contains the context saved before a signal handler is invoked and restored by `sys_rt_sigreturn`.
+#[repr(C)]
+#[repr(align(16))]
+#[derive(Clone, Copy, Debug, Default, Pod)]
+pub struct SigContext {
+    pc: usize,
+    ra: usize,
+    sp: usize,
+    gp: usize,
+    tp: usize,
+    t0: usize,
+    t1: usize,
+    t2: usize,
+    s0: usize,
+    s1: usize,
+    a0: usize,
+    a1: usize,
+    a2: usize,
+    a3: usize,
+    a4: usize,
+    a5: usize,
+    a6: usize,
+    a7: usize,
+    s2: usize,
+    s3: usize,
+    s4: usize,
+    s5: usize,
+    s6: usize,
+    s7: usize,
+    s8: usize,
+    s9: usize,
+    s10: usize,
+    s11: usize,
+    t3: usize,
+    t4: usize,
+    t5: usize,
+    t6: usize,
+    // In RISC-V, the signal stack layout places the FPU context directly
+    // after the general-purpose registers.
+}
+
+impl SigContext {
+    pub fn copy_user_regs_to(&self, dst: &mut UserContext) {
+        let gp_regs = dst.general_regs_mut();
+        copy_gp_regs!(self, gp_regs);
+        dst.set_instruction_pointer(self.pc);
+    }
+
+    pub fn copy_user_regs_from(&mut self, src: &UserContext) {
+        let gp_regs = src.general_regs();
+        copy_gp_regs!(gp_regs, self);
+        self.pc = src.instruction_pointer();
     }
 
-    pub fn copy_from_raw(&mut self, src: &GeneralRegs) {
-        copy_gp_regs!(src, self);
+    pub fn fpu_context_addr(&self) -> Option<Vaddr> {
+        None
     }
+
+    pub fn set_fpu_context_addr(&mut self, _addr: Vaddr) {}
```

Comment:
All uses of `fpu_context_addr`/`set_fpu_context_addr` are under `#[cfg(arch = "x86_64")]`, so I don't think they need to appear for the RISC-V architecture.